crates/task-impls/src/response.rs

```diff
@@ -131,9 +135,20 @@ impl<TYPES: NodeType> NetworkResponseState<TYPES> {
             .is_some_and(|m| m.contains_key(key));
         if !contained {
             let mut consensus = RwLockUpgradableReadGuard::upgrade(consensus).await;
-            consensus
+            if consensus
                 .calculate_and_update_vid(view, Arc::clone(&self.quorum), &self.private_key)
-                .await;
```

Comment:
I think I missed this in a previous pr but I think we probably want to move the calculation of vid shares back outside of consensus.  This forces a write lock to be held to calculate VID which will basically block everything.

I think just the update VID should need the lock